packages/@aws-cdk/aws-ec2/lib/vpn.ts

```diff
@@ -0,0 +1,135 @@
+import cdk = require('@aws-cdk/cdk');
+import { CfnCustomerGateway, CfnVPNConnection, CfnVPNConnectionRoute } from './ec2.generated';
+import { IVpcNetwork } from './vpc-ref';
+
+export interface IVpnConnection extends cdk.IConstruct {
+  /**
+   * The id of the VPN connection.
+   */
+  readonly vpnId: string;
+
+  /**
+   * The id of the customer gateway.
+   */
+  readonly customerGatewayId: string;
+
+  /**
+   * The ip address of the customer gateway.
+   */
+  readonly customerGatewayIp: string;
+
+  /**
+   * The ASN of the customer gateway.
+   */
+  readonly customerGatewayAsn: number;
+}
+
+export interface VpnTunnelOption {
+  /**
+   * The pre-shared key (PSK) to establish initial authentication between the virtual
+   * private gateway and customer gateway.
+   */
+  presharedKey: string;
+
+  /**
+   * The range of inside IP addresses for the tunnel. Any specified CIDR blocks must be
+   * unique across all VPN connections that use the same virtual private gateway.
+   */
+  tunnelInsideCidr: string;
```

Comment:
Both things here are independently optional https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-vpnconnection-vpntunneloptionsspecification.html and so one might want to set one and not the other.

Reply:
Will fix.